swfo/brdf_fallback.py

```diff
@@ -708,69 +712,48 @@ def __get_clean_data(data_filename, data_key, data_window):
-            'doy_average': [],
-        },
-        'properties': {}
-    }
-    start_ds, *_, end_ds = sorted(h5info.keys())
-
-    for datestr, fp in h5info.items():
-        src_md = _reader(fp)
-        curr_doy = datetime.datetime.strptime(datestr, '%Y.%m.%d').strftime('%j')
-        metadata_doc['lineage']['brdf_threshold'].append(src_md['id'])
-        if curr_doy == doy:
-            metadata_doc['lineage']['doy_average'].append(src_md['id'])
-
-        if not metadata_doc.get('grids', None):
-            metadata_doc['crs'] = src_md['crs']
-            metadata_doc['geometry'] = src_md['geometry']
-            metadata_doc['grids'] = src_md['grids']
-            metadata_doc['measurements'] = '???'  # Need to know the measurements
-            metadata_doc['properties'] = {
-                'dtr:start_datetime': get_datetime(
-                    datetime.datetime.strptime(start_ds, '%Y.%m.%d')).isoformat(),
-                'dtr:end_datetime': get_datetime(
-                    datetime.datetime.strptime(end_ds, '%Y.%m.%d')).isoformat(),
-                'eo:instrument': src_md['properties']['eo:instrument'],
-                'eo:platform': src_md['properties']['eo:platform'],
-                'eo:gsd': src_md['properties']['eo:gsd'],
-                'eo:epsg': src_md['properties']['eo:epsg'],
-                'item:providers': [{
-                    'name': 'Geoscience Australia',
-                    'roles': ['processor', 'host'],
-                }],
-                'odc:creation_datetime': get_datetime().isoformat(),
-                'odc:file_format': 'HDF5',
-                'odc:region_code': src_md['properties']['odc:region_code']
-            }
-
-    return metadata_doc
+def munge_metadata(fp, start_ds=None, end_ds=None, only_id=True): 
+    """
+    extracts metadata from MCD43A1 .h5 files. Returns only uuid of the 
+    h5 files if only_id is set to True. Else general metadata attributes 
+    of the h5 file is returned. 
+    """
+    with h5py.File(fp, 'r') as src: 
+        src_md = YAML.load(src[METADATA_OFFSET][()].item())
+        
+        if only_id: 
+            return src_md['id']
+        
+        metadata_doc = { 
+            'id': str(uuid.uuid4()),  # Not sure what the params would be for deterministic uuid
+            'product': {'href': FALLBACK_PRODUCT_HREF},
+            'lineage': {
+                'brdf_threshold': [],
+                'doy_average': []
+                }
+        }
+        metadata_doc['crs'] = src_md['crs']
+        metadata_doc['geometry'] = src_md['geometry']
+        metadata_doc['grids'] = src_md['grids']
+        metadata_doc['measurements'] = '???'  # Need to know the measurements
+        metadata_doc['properties'] = {
+            'dtr:start_datetime': get_datetime(
+                datetime.datetime.strptime(start_ds, '%Y.%m.%d')).isoformat(),
+            'dtr:end_datetime': get_datetime(
+                datetime.datetime.strptime(end_ds, '%Y.%m.%d')).isoformat(),
```

Comment:
I know I stuck these dtr properties in, but as I mentioned in person I now think that providing these fields requires an accepted approach for DEA datasets; I believe it would be more accurate to provide year-{{day-of-year|%m.%d}} as these dates....
However my uncertainty is sufficient to warrant removing these "dtr:*" attributes altogether.